ci/script/functions.sh

```diff
@@ -9,14 +9,20 @@ SPECS_HAVE_RUN_FILE=specs.out
 MAINTENANCE_BRANCH=`cat maintenance-branch`
 
 # Don't allow rubygems to pollute what's loaded. Also, things boot faster
-# without the extra load time of rubygems. Only works on MRI Ruby 1.9+
-if is_mri_192_plus; then
+# without the extra load time of rubygems. Only works on MRI.
+if is_mri; then
   export RUBYOPT="--disable=gem"
 fi
 
 function clone_repo {
   if [ ! -d $1 ]; then # don't clone if the dir is already there
-    travis_retry eval "git clone https://github.com/rspec/$1 --depth 1 --branch $MAINTENANCE_BRANCH"
+    if [ -z "$2" ]; then
+      BRANCH_TO_CLONE="$MAINTENANCE_BRANCH"
```

Comment:
Because `rspec-rails` is alienated from other repos, there's no `4-0-dev` branch there, and we clone `main`.